codegen/src/test/resources/software/amazon/awssdk/codegen/poet/model/PaginatedOperationWithResultKeyPaginator.java

```diff
@@ -0,0 +1,97 @@
+package software.amazon.awssdk.services.jsonprotocoltests.paginators;
+
+import java.util.Iterator;
+import java.util.function.Function;
+import javax.annotation.Generated;
+import software.amazon.awssdk.core.pagination.NextPageFetcher;
+import software.amazon.awssdk.core.pagination.PaginatedItemsIterable;
+import software.amazon.awssdk.core.pagination.PaginatedResponsesIterator;
+import software.amazon.awssdk.core.pagination.SdkIterable;
+import software.amazon.awssdk.services.jsonprotocoltests.JsonProtocolTestsClient;
+import software.amazon.awssdk.services.jsonprotocoltests.model.PaginatedOperationWithResultKeyRequest;
+import software.amazon.awssdk.services.jsonprotocoltests.model.PaginatedOperationWithResultKeyResponse;
+import software.amazon.awssdk.services.jsonprotocoltests.model.SimpleStruct;
+
+/**
+ * Represents the output for the
+ * {@link JsonProtocolTestsClient#paginatedOperationWithResultKeyIterable(PaginatedOperationWithResultKeyRequest)}
+ * operation which is a paginated operation. This class is an iterable of
+ * {@link PaginatedOperationWithResultKeyResponse} that can be used to iterate through all the response pages of the
+ * operation.
+ *
+ * When you make the
+ * {@link JsonProtocolTestsClient#paginatedOperationWithResultKeyIterable(PaginatedOperationWithResultKeyRequest)} API
+ * call, an instance of this class is returned. At this point, no service calls are made yet and so there is no
+ * guarantee that your request is valid. As you iterate through this iterable, SDK will lazily load response pages by
+ * making service calls until there are no pages left or your iteration stops.
+ *
+ * Following are few ways to iterate through the response pages:
+ *
+ * PaginatedOperationWithResultKeyPaginator responses = client.paginatedOperationWithResultKeyIterable(request);
+ *
+ *
+ * 1) Using Streamsresponses.stream().forEach(....);
```

Comment:
Probably worth the duplication of calling the method so we can have each example self contained. 

I.E.
1) Using a Stream
\<pre>
{@code 
PaginatedOperationWithResultKeyPaginator responses = client.paginatedOperationWithResultKeyIterable(request);
responses.stream.forEach(System.out::println);
}
\</pre>

Reply:
sounds good. will do.